### DISCUSSION TITLE
unable to set the rabbitmq queue to lazy queue

### DISCUSSION THREAD

**Discussion by shettypriy** [rabbitmq/rabbitmq-server#4151] [Other] [answered]
I have referred the link https://www.rabbitmq.com/lazy-queues.html here and set the rabbitmq queue to lazy queue using the following command 

rabbitmqctl set_policy Lazy "^lazy-queue$" '{"queue-mode":"lazy"}' --apply-to queues

However, when checking it using the command curl  -u guest:guest 'localhost:15672/api/queues' it still shows the "mode":"default" . How do I set the queue to lazy queue in rabbitmq. Could someone please help

**Comment by mkuratczyk** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by mkuratczyk** [comment, marked as answer]
We can't really help without the exact steps you execute. It works for me:

```
# declare a queue and publish some messages
$ perf-test -y 0 --auto-delete false -f persistent -C 10 -u lazy-queue

# check the queue mode
$ curl -s -uguest:guest 'localhost:15672/api/queues/%2F/lazy-queue' | jq .backing_queue_status.mode
"default"

# set the policy
$ rabbitmqctl set_policy Lazy "^lazy-queue$" '{"queue-mode":"lazy"}' --apply-to queues
Setting policy "Lazy" for pattern "^lazy-queue$" to "{"queue-mode":"lazy"}" with priority "0" for vhost "/" ...

# check the queue mode now
$ curl -s -uguest:guest 'localhost:15672/api/queues/%2F/lazy-queue' | jq .backing_queue_status.mode
"lazy"
```

Perhaps you have another policy that takes precedence?

**Reply by shettypriy** [reply]
We are using the ruby gem 'bunny' in our ruby code to create queues. We have multiple queues created with message published to it. Do we need to enable the lazy queue to each of the queues created? I installed rabbitmq on one of the debian 11 machine and used the command rabbitmqctl set_policy Lazy "^lazy-queue$" '{"queue-mode":"lazy"}' --apply-to queues to enable the lazy queue. Will this not work. Is there any other way to enable lazy queue

**Reply by mkuratczyk** [reply]
`"^lazy-queue$"` is a regular expression to match the queues that should be lazy. This regexp matches only the name "lazy-queue" (because `^` means the beginning of the word and `$` means the end of it). So perhaps the reason it doesn't work for you is that you run this command but then call your queues differently? If you want multiple queues to be lazy, then you need to change the regex to match the names you want.

**Reply by shettypriy** [reply]
okay. Now I get it . Thank you

**Reply by shettypriy** [reply]
I was able to enable the lazy queue for a queue in idle state but not for a queue which is in running state. When I run the following command "rabbitmqctl set_policy Lazy "queue name" '{"queue-mode":"lazy"}' --apply-to queues" it shows the message as below

Setting policy "Lazy" for pattern "queue name" to "{"queue-mode":"lazy"}" with priority "0" for vhost "/" ...

But when I check using the curl command the mode is still set to default for the respective queue .

**Comment by michaelklishin** [comment]
[Policy priority conflicts](https://rabbitmq.com/parameters.html#combining-policy-definitions) is by the most likely reason.

**Comment by bhaskarpc** [comment]
Hi All,

I want to set the lazy queue concept for all my queues, using ruby code how we can set lazy queue. can someone please help?

I was trying like this but it didn't work rmq_channel.queue("test_queue", {:durable => true, :arguments => {"x-queue-mode" => "lazy"}}).

Thanks.

**Reply by michaelklishin** [reply]
Please start new discussions for your questions instead of hijacking existing ones.

**Reply by michaelklishin** [reply]
While you can set queue mode using x-arguments, the recommended way is to [use a policy](https://rabbitmq.com/lazy-queues.html#configuration).

[Policy priority conflicts](https://rabbitmq.com/parameters.html#combining-policy-definitions) is by the most likely reason for policies not being applied.
